Userland/Libraries/LibJS/Runtime/FunctionConstructor.cpp

```diff
@@ -31,54 +32,65 @@ FunctionConstructor::~FunctionConstructor()
 {
 }
 
-Value FunctionConstructor::call()
-{
-    return construct(*this);
-}
-
-Value FunctionConstructor::construct(Function&)
+// 20.2.1.1.1 CreateDynamicFunction ( constructor, newTarget, kind, args ), https://tc39.es/ecma262/#sec-createdynamicfunction
+RefPtr<FunctionExpression> FunctionConstructor::create_dynamic_function_node(GlobalObject& global_object, Function&, FunctionKind kind)
 {
-    auto& vm = this->vm();
+    auto& vm = global_object.vm();
     String parameters_source = "";
     String body_source = "";
     if (vm.argument_count() == 1) {
-        body_source = vm.argument(0).to_string(global_object());
+        body_source = vm.argument(0).to_string(global_object);
         if (vm.exception())
             return {};
     }
     if (vm.argument_count() > 1) {
         Vector<String> parameters;
         for (size_t i = 0; i < vm.argument_count() - 1; ++i) {
-            parameters.append(vm.argument(i).to_string(global_object()));
+            parameters.append(vm.argument(i).to_string(global_object));
             if (vm.exception())
                 return {};
         }
         StringBuilder parameters_builder;
         parameters_builder.join(',', parameters);
         parameters_source = parameters_builder.build();
-        body_source = vm.argument(vm.argument_count() - 1).to_string(global_object());
+        body_source = vm.argument(vm.argument_count() - 1).to_string(global_object);
         if (vm.exception())
             return {};
     }
-    auto source = String::formatted("function anonymous({}\n) {{\n{}\n}}", parameters_source, body_source);
+    auto is_generator = kind == FunctionKind::Generator;
+    auto source = String::formatted("function{} anonymous({}\n) {{\n{}\n}}", is_generator ? "*" : "", parameters_source, body_source);
     auto parser = Parser(Lexer(source));
-    auto function_expression = parser.parse_function_node<FunctionExpression>();
+    auto function = parser.parse_function_node<FunctionExpression>();
     if (parser.has_errors()) {
         auto error = parser.errors()[0];
-        vm.throw_exception<SyntaxError>(global_object(), error.to_string());
+        vm.throw_exception<SyntaxError>(global_object, error.to_string());
         return {};
     }
 
+    return function;
+}
+
+Value FunctionConstructor::call()
+{
+    return construct(*this);
+}
+
+Value FunctionConstructor::construct(Function& new_target)
+{
```

Comment:
Let's add section+link while we're here:

```suggestion
// 20.2.1.1 Function ( p1, p2, … , pn, body ), https://tc39.es/ecma262/#sec-function-p1-p2-pn-body
Value FunctionConstructor::call()
{
    return construct(*this);
}

// 20.2.1.1 Function ( p1, p2, … , pn, body ), https://tc39.es/ecma262/#sec-function-p1-p2-pn-body
Value FunctionConstructor::construct(Function& new_target)
{
```